addons/website_sale_loyalty/models/sale_order.py

```diff
@@ -240,3 +240,17 @@ def _cart_find_product_line(self, product_id, line_id=None, **kwargs):
         lines = super()._cart_find_product_line(product_id, line_id, **kwargs)
         lines = lines.filtered(lambda l: not l.is_reward_line) if not line_id else lines
         return lines
+
+    def _validate_zero_amount_cart(self):
+        """
+        This method overrides the base implementation from the `website_sale` module.
+        Validates the cart when the total amount is zero and creates an invoice if applicable.
+
+        :return: True if an invoice is created and posted, otherwise False.
+        """
+        super()._validate_zero_amount_cart()
+        auto_invoice = self.env['ir.config_parameter'].get_param('sale.automatic_invoice')
+        if auto_invoice and self.reward_amount:
```

Comment:
My bad, I should have noticed earlier, but auto_invoice could be a string value representing False, and it would be True if you check it like that.
You should ideally use `str2bool` util to make sure falsy values won't be considered truthy.